packages/esm-service-queues-app/src/queue-services/queue-service-form.workspace.tsx

```diff
@@ -1,180 +1,174 @@
-import React, { useCallback, useState } from 'react';
+import React from 'react';
 import { useTranslation } from 'react-i18next';
+import { useForm, Controller } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+
 import {
   Button,
   ButtonSet,
   Column,
   Form,
-  InlineNotification,
   Layer,
   Select,
   SelectItem,
   Stack,
   TextInput,
+  InlineLoading,
 } from '@carbon/react';
 import { mutate } from 'swr';
 import { type DefaultWorkspaceProps, restBaseUrl, showSnackbar } from '@openmrs/esm-framework';
 import { saveQueue, useServiceConcepts } from './queue-service.resource';
 import { useQueueLocations } from '../create-queue-entry/hooks/useQueueLocations';
 import styles from './queue-service-form.scss';
+import { t } from 'i18next';
+import type { TFunction } from 'i18next';
+
+const createQueueServiceSchema = (t: TFunction) =>
+  z.object({
+    queueName: z
+      .string({
+        required_error: t('queueNameRequired', 'Queue name is required'),
+      })
+      .trim()
+      .min(1, t('queueNameRequired', 'Queue name is required')),
+    queueConcept: z.string({
+      required_error: t('queueConceptRequired', 'Queue concept is required'),
+    }),
+    userLocation: z.string({
+      required_error: t('queueLocationRequired', 'Queue location is required'),
+    }),
+  });
```

Comment:
```suggestion
const createQueueServiceSchema = (t: TFunction) =>
  z.object({
    queueName: z
      .string({
        required_error: t('queueNameRequired', 'Queue name is required'),
      })
      .trim()
      .min(1, t('queueNameRequired', 'Queue name is required')),
    queueConcept: z.string({
      required_error: t('queueConceptRequired', 'Queue concept is required'),
    }),
    userLocation: z.string({
      required_error: t('queueLocationRequired', 'Queue location is required'),
    }),
  });
```

Reply:
it looks like same any changes @denniskigen

Reply:
const createQueueServiceSchema = (t: TFunction) =>
  z.object({
    queueName: z
      .string()
      .trim()
      .nonempty({ message: t('queueNameRequired', 'Queue name is required') }),
    queueConcept: z
      .string()
      .nonempty({ message: t('queueConceptRequired', 'Queue concept is required') }),
    userLocation: z
      .string()
      .nonempty({ message: t('queueLocationRequired', 'Queue location is required') }),
  });

Reply:
You're right. I've updated the suggestion above.